### ISSUE TITLE
MarshalString() should not escape utf8 characters. Because php does not support escaping in serializing

### ISSUE THREAD

**Issue by szmcdull** [elliotchance/phpserialize#5]
```
$ cat test2.php
<?php
$s = unserialize('s:3:"\xe6\x82\xa8";');
var_dump($s);
?>
$ php test2.php
PHP Notice:  unserialize(): Error at offset 8 of 19 bytes in /data_log/neiwang81/test2.php on line 2
bool(false)
```

**Comment by elliotchance** [comment]
What's the behaviour of the Go package?

**Comment by szmcdull** [comment]
If you mean the phpserialize package, it escape all runes above 127, which is not expected by PHP:

```
func MarshalString(value string) []byte {
	var buffer bytes.Buffer
	for _, r := range value {
		if r <= 127 {
		buffer.WriteRune(r)
		} else {
			chars := make([]byte, 4)
			for i := 0; i < utf8.EncodeRune(chars, r); i++ {
				escapedChar := fmt.Sprintf("\\x%02x", chars[i])
				buffer.WriteString(escapedChar)
			}
		}
	}

	return []byte(fmt.Sprintf("s:%d:\"%s\";", len(value), buffer.String()))
}
```

**Comment by szmcdull** [comment]
If i remove the if else blocks and write directly to the buffer, it works all right:

```
func MarshalString(value string) []byte {
	var buffer bytes.Buffer
	for _, r := range value {
		buffer.WriteRune(r)
	}

	return []byte(fmt.Sprintf("s:%d:\"%s\";", len(value), buffer.String()))
}
```

**Comment by elliotchance** [comment]
Thanks @szmcdull, welcoming a PR...

**Comment by szmcdull** [comment]
I wonder why you escape it in the first place

**Comment by elliotchance** [comment]
Fixed in v1.1.2.
